src/game/common/rts/player.cpp

```diff
@@ -100,25 +120,476 @@ Player::~Player()
+        m_aiSquad = nullptr;
+    }
+
+    m_aiSquad = new Squad();
+
+    if (m_tunnelSystem != nullptr) {
+        m_tunnelSystem->Delete_Instance();
+        m_tunnelSystem = nullptr;
+    }
+
+    m_canBuildBase = true;
+    m_canBuildUnits = true;
+    m_playerIsObserver = false;
+    m_bountyCostToBuild = 0.0f;
+    m_listInScoreScreen = true;
+    m_unitsShouldHunt = false;
+#ifdef GAME_DEBUG_STRUCTS
+    m_ignorePrereqs = false;
+    m_freeBuild = false;
+    m_instantBuild = false;
 #endif
+
+    if (pt != nullptr) {
+        m_side = pt->Get_Side_Name();
+        m_baseSide = pt->Get_Base_Side();
+        m_productionCostChanges = *pt->Get_Production_Cost_Changes();
+        m_productionTimeChanges = *pt->Get_Production_Time_Changes();
+        m_productionVeterancyLevels = *pt->Get_Production_Veterancy_Levels();
+        m_playerColor = pt->Get_Preferred_Color()->Get_As_Int() | 0xFF000000;
+        m_playerNightColor = m_playerColor;
+        m_money = *pt->Get_Money();
+        m_money.Set_Player_Index(Get_Player_Index());
+        m_handicap = *pt->Get_Handicap();
+
+        if (m_money.Get() == 0) {
+            if (g_theGameInfo != nullptr) {
+                m_money = *g_theGameInfo->Get_Money();
+            } else {
+                m_money = g_theWriteableGlobalData->m_defaultStartingCash;
+            }
+        }
+
+        m_playerDisplayName.Clear();
+        m_playerName.Clear();
+        m_playerNameKey = NAMEKEY_INVALID;
+        m_playerType = PLAYER_COMPUTER;
+        m_playerIsObserver = pt->Is_Observer();
+        m_playerIsDead = m_playerIsObserver;
+    } else {
+        m_side = "";
+        m_baseSide = "";
+        m_productionCostChanges.clear();
+        m_productionTimeChanges.clear();
+        m_productionVeterancyLevels.clear();
+        m_playerColor = -1;
+        m_playerNightColor = -1;
+        m_money.Empty();
+        m_handicap.Init();
+        m_playerDisplayName = Utf16String::s_emptyString;
+        m_playerName = Utf8String::s_emptyString;
+        m_playerNameKey = g_theNameKeyGenerator->Name_To_Key(Utf8String::s_emptyString.Str());
+        m_playerType = PLAYER_COMPUTER;
+        Set_Player_Relationship(this, ALLIES);
+    }
+
+    m_scoreKeeper.Reset(m_playerIndex);
+    m_playerTemplate = pt;
+    Reset_Rank();
+    m_disabledSciences.clear();
+    m_hiddenSciences.clear();
+
+    for (auto i = m_specialPowerReadyTimerList.begin(); i != m_specialPowerReadyTimerList.end();) {
+        SpecialPowerReadyTimerType &timer = *i;
+        i = m_specialPowerReadyTimerList.erase(i);
+        timer.Reset();
+    }
+
+    for (auto i = m_kindOfPercentProductionChangeList.begin(); i != m_kindOfPercentProductionChangeList.end();) {
+        KindOfPercentProductionChange *k = (*i);
+        i = m_kindOfPercentProductionChangeList.erase(i);
+
+        if (k != nullptr) {
+            k->Delete_Instance();
+        }
+    }
+
+    Get_Academy_Stats()->Init(this);
+    m_retaliationModeEnabled = false;
```

Comment:
It appears `m_multiplayerStartIndex` remains uninitialized at this point.

Reply:
Its initialized later on by Player::Init_From_Dict